master/internal/task/allocation.go

```diff
@@ -961,7 +968,10 @@ func (a *Allocation) terminated(ctx *actor.Context, reason string) {
 	if a.exitErr != nil {
 		level = ptrs.Ptr(model.LogLevelError)
 	}
-	defer a.sendEvent(ctx, sproto.Event{Level: level, ExitedEvent: &exitReason})
+	defer a.sendTaskLog(ctx, a.enrichLog(model.TaskLog{
+		Level: level,
+		Log:   fmt.Sprintf("%s was terminated: %s", a.req.Name, exitReason),
```

Comment:
i think your e2e issue is caused by the swap from pointer to non pointer exitReason here

Reply:
since the `a.enrichLog` call is evaluated right away with the default reason from `961`. there is some tricky defer here, sorry.

Reply:
To clarify: when you defer a function call, the arguments are evaluated at the time of deferral. If you want the logic which happens later to reflect changes made to variables, you'll want to either use a pointer or defer an anonymous function which closes over the variable [like this](https://go.dev/play/p/Uu81JCucM6U).

Reply:
Hm the example you show doesn't seem super readable -- so should I go back & fix the deferred call, or is the output, as it looks now, okay?

Reply:
Fixed with new commit -- wrapped the sendTaskLog call in a function